easybuild/easyconfigs/w/waLBerla/waLBerla-6.1-foss-2022b.eb

```diff
@@ -0,0 +1,56 @@
+# easyconfig file for waLBerla v6.1
+easyblock = 'CMakeMakeCp'
+
+name = 'waLBerla'
+version = '6.1'
+
+homepage = "https://walberla.net/index.html"
+description = """Widely applicable Lattics-Boltzmann from Erlangen is a 
+block-structured high-performance framework for multiphysics simulations"""
+
+toolchain = {'name': 'foss', 'version': '2022b'}
+toolchainopts = {'usempi': True, 'pic': True}
+
+source_urls = ['https://i10git.cs.fau.de/walberla/walberla/-/archive/v6.1']
+sources = ['%(name)s-v%(version)s.tar.gz']
+patches = ['waLBerla-6.1_fix_cmakelist_for_easybuild_without_python.patch']
+checksums = [
+    {'waLBerla-v6.1.tar.gz': 'f0acdd9ad6543bc9306c8aae953dd5065986271d4398916ae0469db8b21c007a'},
+    {'waLBerla-6.1_fix_cmakelist_for_easybuild_without_python.patch':
+     'e64927d85ed6ac4c52921c33c504d4f6e7038fd618c89cf7efdadbcf638a1048'},
+]
+
+builddependencies = [('CMake', '3.24.3')]
+
+dependencies = [
+    ('Boost.MPI', '1.81.0'),
+]
+
+parallel = 1
+
+configopts = "-DWALBERLA_BUILD_WITH_PYTHON=OFF "
+configopts += "-DWALBERLA_BUILD_SHOWCASES=OFF "
+configopts += "-DWALBERLA_BUILD_DOC=OFF "
+configopts += "-DWALBERLA_BUILD_BENCHMARKS=OFF"
+
+files_to_copy = [
+    (['%(builddir)s/easybuild_obj/src/*'], 'src'),
```

Comment:
Aren't these header files? Shouldn't they be installed in `include`

Reply:
It's not just header files, it's a bunch of libraries too, for example:

```
src/gather/libgather.a
```

It's not exactly a clean installation though, since it also includes a bunch of `.o` files and `.cmake` files.

That said, I think just copying everything to `src` makes sense, waLBerla is a bit of an odd duck when it comes to installing it...

Reply:
Yes, it's not very clear cut... I kept things in `src` assuming that's where users are used to finding the files already. But I also realise I made a mistake, there are two `src` directories one ends up with if following the installation directions [here](https://www.walberla.net/doxygen/setup_instructions.html): 
- one inside `build`, which indeed has also a bunch of `.o` and `.cmake` files, plus some source files 
- in the unpacked source directory (obviously) which contains more `source` files and most of the header files.

For this particular case I think we'd also want to copy the other sources and end up with something like `build/*/` (which includes `src` and other directories, including `.o` and `.cmake` files) and `src/` which has a very similar structure but includes most of the `.h` and other source files. I hope this makes sense, pushing the changes now.